linked_list/linked_list.rb

```diff
@@ -46,14 +46,32 @@ def remove_item(data)
       prev = current
       current = current.next
     end
+  end
 
-    def search_data(data)
-      raise "We cannot delete empty linked list" if @head == nil
-      current = @head
-      while current
-        return "element exists" if current.data == data
-      end
-      "can't find data in linked list"
+  def search_data(data)
+    raise "We cannot delete empty linked list" if @head == nil
+    current = @head
+    while current
+      return "element exists" if current.data == data
     end
+    "can't find data in linked list"
   end
+
+end
+
+class DisplayLinkedList
```

Comment:
Okay let's create a new class in new file and implement printing specifically for it?
I think it will be much cleaner. Also, also lets have base class with abstract method ```output```